.github/workflows/auto-cherry-picker.yaml

```diff
@@ -14,23 +14,97 @@ on:
         required: true
 
 jobs:
-  picker:
+  prerequisites:
+    name: Gather Prerequisites
     if: (github.event.pull_request.merged == true && contains(github.event.pull_request.labels.*.name, 'needs-cherrypick') ) || github.event_name == 'workflow_dispatch'
     runs-on: ubuntu-latest
+    outputs:
+      milestones: ${{ steps.get-prereqs.outputs.milestones }}
+      merge_commit: ${{ steps.get-prereqs.outputs.merge_commit }}
+      pr_num: ${{ steps.get-prereqs.outputs.pr_num }}
+    steps:
+      - id: get-prereqs
+        name: Get Cherry-Pick prerequisites
+        env:
+          PR_NUM: ${{ github.event.pull_request.number || inputs.PR_number }}
+        run: |
+          TARGET_MILESTONE=$(gh pr view "$PR_NUM" --json milestone --jq '.milestone.title')
+          RELEVANT_MILESTONES=$(gh api graphql -F owner=":owner" -F name=":repo" -f query='
+            query ListMilestones($name: String!, $owner: String!) {
+              repository(owner: $owner, name: $name) {
+                milestones(last: 10) {
+                  nodes {title}
+                }
+              }
+            }' --jq .data.repository.milestones.nodes.[].title | sort -V | awk "/$TARGET_MILESTONE/{p=1}p" -)
+
+          echo "milestones=[$(RELEVANT_MILESTONES | tr '\n' ',')]" >> "$GITHUB_OUTPUT"
+
+          MERGE_COMMIT=$(gh pr view "$PR_NUM" --json mergeCommit --jq '.mergeCommit.oid')
+          echo "merge_commit=$MERGE_COMMIT" >> "$GITHUB_OUTPUT"
+
+          echo "pr_num=$PR_NUM" >> "$GITHUB_OUTPUT"
+
+  picker:
+    name: Cherry-Pick
+    needs: prerequisites
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        milestone: ${{ needs.prerequisites.outputs.milestones }}
```

Comment:
That makes sense. I assume the "type" of the expression is a string. That doesn't matter when running steps because the code is a string but here in the evaluation context it needs to be more strongly typed.

That being said I'll try without and then if it goes boom, add it.